### PR TITLE
ci: add pre-commit for toml sorting, fix maintainers listing

### PR DESCRIPTION
**Pull request by JacobCoffee**
## Description

- Adds a new pre-commit entry for toml and yaml sorting
- Updates maintainers list because the new ones were lazy 😛

### CODE REVIEW

**Comment by JacobCoffee** [comment]
The reason behind the toml sorter thing is because
1. I found it the other day, its pretty cool. I dont like how it sorts, but I like that its the same.
2. People are always committing PRs with changes to this because *their* formatter hits it. This will help clear that diff clutter

File: pyproject.toml
Comment (JacobCoffee) [inline pyproject.toml:1]: maybe its not so great idk what the hell it did to the keys

File: pyproject.toml [hunk 1]
```diff
@@ -1,10 +1,73 @@
...
+]
+"tests/unit/test_contrib/test_sqlalchemy/**/*.*" = ["UP006"]
+"tests/unit/test_openapi/test_typescript_converter/test_converter.py" = ["W293"]
+"tools/**/*.*" = ["D", "ARG", "EM", "TRY", "G", "FBT"]
+"tools/prepare_release.py" = ["S603", "S607"]
+
+[lint.pydocstyle]
+convention = "google"
+
 [project]
```
Comment (guacs) [inline pyproject.toml:64]: Is there some way to get the `project` section to be the first section in the file? Not a huge deal, but I just felt it probably makes sense to keep it as the first section.

Comment (JacobCoffee) [reply pyproject.toml:64]: idk about this, i absolutely hate how it sorts everything, but having it consistent is cool. Ill look into it if i can solve it removing `[tool.ruff.` from a bunch of keys..

File: pyproject.toml [hunk 2]
```diff
@@ -174,28 +298,27 @@ docs-serve = "sphinx-autobuild docs docs/_build/ -j auto --watch litestar --watc
...
+  "test_apps",
+  "tools",
+  "docs",
+  "tests/examples",
+  "tests/docker_service_fixtures.py",
+  "litestar/contrib/pydantic/pydantic_dto_factory.py",
+  "litestar/contrib/pydantic/pydantic_init_plugin.py",
+  "tests/unit/test_contrib/test_sqlalchemy.py"
 ]
-fail_under = 96
```
Comment (JacobCoffee) [inline pyproject.toml:198]: is this thing on crack it just moves keys out from under sections to random ones

**Comment by JacobCoffee** [comment]
this thing is stupid

**Pull request closed without merging**